### ISSUE TITLE
Is there a rule to format associated value bindings in enums?

### ISSUE THREAD

**Issue by tgrapperon** [nicklockwood/SwiftFormat#981]
Hello,

When an enum case has associated values, one can bind them to variables with `let` by:
* Prefixing the case name: `case let .format(string):…`
* Repeating for each value: `case .format(let string):…`

Is there an existing rule in SwiftFormat to specify that all `let` should be in prefix position for example?
Thanks!

**Comment by nicklockwood** [comment]
@tgrapperon yes, the rule for this is `hoistPatternLet` (and it's enabled by default).

**Comment by tgrapperon** [comment]
Thanks Nick! If I'm trying to format this code:
```swift
enum Action {
  case perform(String)
}

func execute(action: Action) {
  switch action {
  case .perform(let string): // <<<
    print(string)
  }
}
```
I would expect the marked line to be formatted as `case let .perform(string):` when the rule is active, but it does nothing (using SwiftFormat 0.47.9 (1)). Is there any reason/workaround?

**Comment by nicklockwood** [comment]
@tgrapperon  it works for me when I test switformat 0.48.9 (was 0.47/9 a typo?) on a file with those exact contents.

Try running swiftformat from the terminal and seeing if there is any suspicious output - e.g. it may be picking up a .swiftformat config file you didn't intend it to. You can run with the --verbose argument for additional detail about what it's doing.

**Comment by tgrapperon** [comment]
I had an old version of the Xcode plugin (copy/pasted the version's number from "About..." screen) and I've just updated to 0.48.11 and restarted Xcode (13b3).
It works on command line, but using the Xcode plugin, but it seems to kick off only when there are non-homogenous uses:
```swift
case .perform1(let string): break
case .perform2(let string): break
```
is not formatted, like the version with only one case, but
```swift
case .perform1(let string): break
case let .perform2(string): break
```
is formatted as 
```swift
case let .perform1(string): break
case let .perform2(string): break
```
as expected. This doesn't format however if I'm disabling `hoistPatternLet` in the app, so I think that shows that Xcode sees the correct config. I've also reset Xcode rules to their defaults, to no avail.
In other words, with the same default config and the same version, SwiftFormat seems to produce different results in Xcode and command line for this configuration.

**Comment by nicklockwood** [comment]
@tgrapperon the Xcode plugin can't read your .swiftformat config due to sandboxing restrictions.

The default behavior for the plugin is to try to infer the configuration to use by looking at the file being formatted, however you can load a specific configuration to use instead in the Swiftformat for Xcode app. In the app, go to `File > Open...` and load the `.swiftformat` file from your project, then it should be consistent.

**Comment by tgrapperon** [comment]
I'm not using `.swiftformat` files in the projects, as I'm generally using the config provided by the SwiftFormat extension app.
What I'm saying is Xcode is responding to configuration changes made in the app:
|                                       | Command Line, hoistPatternLet | Xcode, hoistPatternLet | Xcode, !hoistPatternLet |
|---------------------------------------|--------------|------------------------|-------------------------|
| case .a(let s):... case .b(let s):... | OK (Format)  | NO                     | OK (no Format)          |
| case .a(let s):... case let .b(s):... | OK (Format)  | OK (Format)            | OK (no Format)          |

The last column shows that Xcode sees `hoistPatternLet` being deactivated (it behaves differently of the mixed case).
I'll fork the repository and try to see what's happening eventually.

**Comment by nicklockwood** [comment]
@tgrapperon is the "Infer Options Automatically" checkbox enabled in the SwiftFormat for Xcode app? Because that would explain it.

**Comment by tgrapperon** [comment]
Indeed! And unchecking solves the issue!
I find it a little surprising, as the option didn't appear  under the influence of automatic inference to me (I thought it concerned only options that are disabled when inference is on).
Thanks!
